freud/density.pyx

```diff
@@ -528,19 +515,29 @@ cdef class GaussianDensity(Compute):
             The image grid with the Gaussian density.
     """  # noqa: E501
     cdef freud._density.GaussianDensity * thisptr
-    cdef arglist
-
-    def __cinit__(self, *args):
-        if len(args) == 3:
-            self.thisptr = new freud._density.GaussianDensity(
-                args[0], args[1], args[2])
-            self.arglist = [args[0], args[1], args[2]]
-        elif len(args) == 5:
-            self.thisptr = new freud._density.GaussianDensity(
-                args[0], args[1], args[2], args[3], args[4])
-            self.arglist = [args[0], args[1], args[2], args[3], args[4]]
+    cdef float r_max
+
+    def __cinit__(self, width, r_max, sigma):
+        cdef vec3[uint] width_vector
+        if isinstance(width, int):
+            width_vector = vec3[uint](width, width, width)
+        elif isinstance(width, (list, tuple)):
```

Comment:
Can we use `Sequence` from `collections.abc`? https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes

Reply:
I actually considered this, but I decided I wanted to be a bit more proscriptive with what we permit. Supporting arbitrary sequences here provides no real benefit IMO, so I would prefer not to be super ducktyping friendly in this context. We could always add that support later if we found a compelling reason to do so.

Reply:
@vyasr If we don't, this will fail with NumPy arrays of input, which I think is a problem. We've made strides to use NumPy arrays explicitly for all list-like numeric inputs and outputs elsewhere in the package.

Reply:
That's fair. Sequence still seems a bit too permissive to me, but not a huge deal.